File: Algorithms/p287_Find_the_Duplicate_Number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int p = 1;
        int q = nums.size()-1;
        while(p<q){
            int mid = (p+q)/2;
            int cnt = 0;
            for(int v : nums){
                if(v<=mid){
                    cnt++;
                }
            }
            if(cnt<=mid){
                p = mid+1;
            }else{
                q = mid;
            }
        }
        return p;
    }
};

File: Algorithms/p287_Find_the_Duplicate_Number/p287_Find_the_Duplicate_Number.py
class Solution(object):
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        p = 1
        q = len(nums)-1
        while p<q:
            mid = (p+q)//2
            cnt = 0
            for v in nums:
                if v<=mid:
                    cnt += 1
            if cnt<=mid:
                p = mid+1
            else:
                q = mid
        return p
